src/ParametersHandler/include/BipedalLocomotion/ParametersHandler/IParametersHandler.h

```diff
@@ -191,7 +208,7 @@ class IParametersHandler
      * check YarpImplementation::setGroup
      * @return true/false in case of success/failure
      */
-    virtual bool setGroup(const std::string& name, shared_ptr newGroup) = 0;
+    virtual bool setGroup(const std::string& name, const_shared_ptr newGroup) = 0;
```

Comment:
The same for the YARP implementation. I would document that the content of the input pointer is cloned and hence any modification to the input pointer afterward will not have an effect.